core/shared/src/main/scala/org/log4s/Logger.scala

```diff
@@ -45,7 +45,7 @@ object Logger {
   }
 }
 
-final class Logger(val logger: JLogger) extends AnyVal {
+final class Logger(val logger: JLogger) extends AnyVal with LoggerCompat {
```

Comment:
In this encoding, the `LoggerCompat` interfaces have a chance to diverge.  Maybe we could self-type a `LoggerContract` trait or something.

Reply:
Have you checked the generated bytecode to make sure this doesn't add any new overhead? My experience in the past is that `AnyVal` traits have a tendency to add layers of indirection.

Reply:
I have not.  I'll check, and fall back to copy and paste if it looks different.

Reply:
Looks the same to me.

```
  public boolean isDebugEnabled();
    descriptor: ()Z
    flags: ACC_PUBLIC
    Code:
      stack=2, locals=1, args_size=1
         0: getstatic     #55                 // Field org/log4s/Logger$.MODULE$:Lorg/log4s/Logger$;
         3: aload_0
         4: invokevirtual #123                // Method logger:()Lorg/slf4j/Logger;
         7: invokevirtual #100                // Method org/log4s/Logger$.isDebugEnabled$extension:(Lorg/slf4j/Logger;)Z
        10: ireturn
      LineNumberTable:
        line 54: 0
      LocalVariableTable:
        Start  Length  Slot  Name   Signature
            0      11     0  this   Lorg/log4s/Logger;
```

Without:

```
  public boolean isDebugEnabled();
    descriptor: ()Z
    flags: ACC_PUBLIC
    Code:
      stack=2, locals=1, args_size=1
         0: getstatic     #53                 // Field org/log4s/Logger$.MODULE$:Lorg/log4s/Logger$;
         3: aload_0
         4: invokevirtual #121                // Method logger:()Lorg/slf4j/Logger;
         7: invokevirtual #98                 // Method org/log4s/Logger$.isDebugEnabled$extension:(Lorg/slf4j/Logger;)Z
        10: ireturn
      LineNumberTable:
        line 54: 0
      LocalVariableTable:
        Start  Length  Slot  Name   Signature
            0      11     0  this   Lorg/log4s/Logger;
```

It adds an invokestatic call to `LoggerCompat` in the constructor, but otherwise looks the same.